examples/llm_inference/ios/InferenceExample/View Models/ConversationViewModel.swift

```diff
@@ -0,0 +1,326 @@
+// Copyright 2024 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+@MainActor
+class ConversationViewModel: ObservableObject {
+  /// This array holds both the view models responsible for presentation of both user and system messages
+  @Published var messageViewModels = [MessageViewModel]()
+
+  /// .error state is set only when there are errors in loading the model and creating a new chat session.
+  /// Response generation errors are relayed as messages from the model.
+  enum State: Equatable {
+    case loadingModel
+    case promptSubmitted
+    case streamingResponse
+    case error
```

Comment:
Can we make this case take in an InferenceError? For example, case failed(error: InferenceError). Then we can just keep the currentState property and remove criticalError and createChatError?